### PR TITLE
Fix bugs in monitoring service

### PR DESCRIPTION
**Pull request by yinan-symphony**
Some bugs detected in certain specific scenarios in monitoring service since the version has been involved.They are fixed in this commit.

### Description
Please put here the intent of your pull request.

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [ ] Referenced a ticket in the PR title or description
- [ ] Filled properly the description and dependencies, if any
- [ ] Unit/Integration tests updated or added
- [ ] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/converter/WorkflowInstDomainVersionConverter.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    implements BiConverter<HistoricProcessInstanceEntity, Map<String, String>, WorkflowInstanceDomain> {
+
+  @Override
+  public WorkflowInstanceDomain apply(HistoricProcessInstanceEntity hisProcInstance,
+      Map<String, String> procIdVersionTagMap) {
+    if (!procIdVersionTagMap.containsKey(hisProcInstance.getProcessDefinitionId())) {
+      return null;
+    }
+    Optional<String> version = Optional.ofNullable(procIdVersionTagMap.get(hisProcInstance.getProcessDefinitionId()));
+    return this.instanceCommonBuilder(hisProcInstance).version(version.map(Long::valueOf).orElse(null)).build();
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/converter/WorkflowInstDomainVersionConverter.java:23]:
```suggestion
Optional<String> version = Optional.ofNullable(procIdVersionTagMap.get(hisProcInstance.getProcessDefinitionId()));

if (version.isPresent()) {
  return this.instanceCommonBuilder(hisProcInstance).version(version.map(Long::valueOf).orElse(null)).build();
} else {
  return null;
}
```

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java [hunk 2]
```diff
@@ -34,30 +34,22 @@ public class WorkflowManagementService {
...
+    String deploy = null;
+    if (workflow.isToPublish()) {
+      deploy = workflowEngine.deploy(context);
+      setCurrentActiveVersionToInactive(workflow.getId());
+    }
+    VersionedWorkflow versionedWorkflow = toVersionedWorkflow(workflow, swadlView, deploy);
+    versionRepository.save(versionedWorkflow);
+  }
+
+  private void checkIfNoPublishedVersionExists(Workflow workflow) {
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java:47]: The method name is a bit confusing for me. It throws an error if the workflow is not found in the versionRepository or if it is found there but the "published" field is set to False.  I would name it **checkPublishedVersionExists** rather. WDYT?

**Review by symphony-soufiane — APPROVED** [review]
Few comments, but lgtm

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java (hunk 2 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java:47]: the check is to make sure there is no workflow having the same id with 'published' field as false, if there is any, it throws an exception. We are not allowing to create a new version while there exists a non-published version.
'checkPublishedVersionExists' does not reflect this meaning.

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/converter/WorkflowInstDomainVersionConverter.java (hunk 1 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/converter/WorkflowInstDomainVersionConverter.java]: Overlooked this refactoring.
code changed now

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java (hunk 2 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java]: I changed again the name, hope it is a bit more clearer

Comment (symphony-soufiane) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java]: Cool thank you!

**Pull request merged**